commit 360f352c9a6da545d798ef3015e73ca114f0d230
Author: Florian Weimer <fweimer@redhat.com>
Date:   Fri Jun 19 18:22:20 2026 +0200

    resolv: Declare __p_class_syms, __p_type_syms for internal use
    
    Reviewed-by: Carlos O'Donell <carlos@redhat.com>
    Reviewed-by: Adhemerval Zanella <adhemerval.zanella@linaro.org>

Conflicts:
	include/resolv.h
	  Context difference
	resolv/res_debug.c
	  Line numbers

diff --git a/include/resolv.h b/include/resolv.h
index 4dbbac3800..d5ad9994b9 100644
--- a/include/resolv.h
+++ b/include/resolv.h
@@ -66,5 +66,10 @@ libresolv_hidden_proto (__res_queriesmat
 libresolv_hidden_proto (__b64_ntop)
 libresolv_hidden_proto (__dn_count_labels)
 
+extern const struct res_sym __p_class_syms[];
+libresolv_hidden_proto (__p_class_syms)
+extern const struct res_sym __p_type_syms[];
+libresolv_hidden_proto (__p_type_syms)
+
 # endif /* _RESOLV_H_ && !_ISOMAC */
 #endif
diff --git a/resolv/res_debug.c b/resolv/res_debug.c
index 73af0c72fe..6bf9962916 100644
--- a/resolv/res_debug.c
+++ b/resolv/res_debug.c
@@ -391,8 +391,6 @@ p_fqname(const u_char *cp, const u_char *msg, FILE *file) {
  * that C_ANY is a qclass but not a class.  (You can ask for records of class
  * C_ANY, but you can't have any records of that class in the database.)
  */
-extern const struct res_sym __p_class_syms[];
-libresolv_hidden_proto (__p_class_syms)
 const struct res_sym __p_class_syms[] = {
   {C_IN,    (char *) "IN"},
   {C_CHAOS, (char *) "CHAOS"},
@@ -427,8 +427,6 @@ const struct res_sym __p_update_section_syms[] attribute_hidden = {
  * Names of RR types and qtypes.  The list is incomplete because its
  * size is part of the ABI.
  */
-extern const struct res_sym __p_type_syms[];
-libresolv_hidden_proto (__p_type_syms)
 const struct res_sym __p_type_syms[] = {
   {ns_t_a,      (char *) "A",     (char *) "address"},
   {ns_t_ns,     (char *) "NS",    (char *) "name server"},
